core/trino-main/src/main/java/io/trino/execution/CreateTableTask.java

```diff
@@ -236,26 +236,29 @@ else if (element instanceof LikeClause) {
                 throw new TrinoException(GENERIC_INTERNAL_ERROR, "Invalid TableElement: " + element.getClass().getName());
             }
         }
-
-        Map<String, Expression> sqlProperties = mapFromProperties(statement.getProperties());
+        if (hasDefaultProperty(statement.getProperties())) {
+            throw semanticException(NOT_SUPPORTED, statement, "Setting a table property to DEFAULT in a CREATE TABLE statement is not supported");
+        }
         Map<String, Object> properties = metadata.getTablePropertyManager().getProperties(
                 catalogName,
                 tableName.getCatalogName(),
-                sqlProperties,
+                statement.getProperties(),
                 session,
                 metadata,
                 accessControl,
-                parameterLookup,
-                true);
+                parameterLookup);
 
-        if (!disableSetPropertiesSecurityCheckForCreateDdl && !properties.isEmpty()) {
+        if (!disableSetPropertiesSecurityCheckForCreateDdl) {
             accessControl.checkCanCreateTable(session.toSecurityContext(), tableName, properties);
         }
         else {
             accessControl.checkCanCreateTable(session.toSecurityContext(), tableName);
         }
 
-        Map<String, Object> finalProperties = combineProperties(sqlProperties.keySet(), properties, inheritedProperties);
+        Set<String> specifiedPropertyKeys = statement.getProperties().stream()
```

Comment:
nit: you could move that to `combineProperties` since it's only `combineProperties` usage